File: app/src/main/java/com/example/project_note/RecyclerViewItemTouchHelper.java
package com.example.project_note;

import android.graphics.Canvas;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.RecyclerView;


/*
* gọi ItemTouchHelper.SimpleCallback  lắng nghe sự kiện "di chuyển" và "vuốt"
* */
public class RecyclerViewItemTouchHelper extends ItemTouchHelper.SimpleCallback {

    private ItemTouchHelperListener mlistener;


    //dragDirs Hướng kéo
    // swipe Dirs Hướng vuốt
    public RecyclerViewItemTouchHelper(int dragDirs, int swipeDirs, ItemTouchHelperListener listener) {

        super(dragDirs, swipeDirs);
        this.mlistener = listener;
    }

    @Override
    public boolean onMove(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull RecyclerView.ViewHolder target) {
        return false;
    }

    // Thông báo mListener khi swiped để thực hiện khi swip
    @Override
    public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {

        if(mlistener != null){
            mlistener.onSwiped(viewHolder);
        }
    }

    //Được gọi khi một item được chọn.
    @Override
    public void onSelectedChanged(@Nullable RecyclerView.ViewHolder viewHolder, int actionState) {
        if(viewHolder != null){
            View foreGroundView = ((NoteAdapter.NoteViewHodel) viewHolder).mLayoutForceGround;
            getDefaultUIUtil().onSelected(foreGroundView);

        }
    }

    //Được gọi khi một item được vẽ trên cùng của các item khác.
    @Override
    public void onChildDrawOver(@NonNull Canvas c, @NonNull RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, float dX, float dY, int actionState, boolean isCurrentlyActive) {
        View foreGroundView = ((NoteAdapter.NoteViewHodel) viewHolder).mLayoutForceGround;
        getDefaultUIUtil().onDrawOver(c, recyclerView, foreGroundView, dX, dY, actionState, isCurrentlyActive);
    }

    //Được gọi khi một item được vẽ.
    @Override
    public void onChildDraw(@NonNull Canvas c, @NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, float dX, float dY, int actionState, boolean isCurrentlyActive) {
        View foreGroundView = ((NoteAdapter.NoteViewHodel) viewHolder).mLayoutForceGround;
        getDefaultUIUtil().onDraw(c, recyclerView, foreGroundView, dX, dY, actionState, isCurrentlyActive);
    }

    //Được gọi khi việc vẽ item kết thúc
    @Override
    public void clearView(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder) {
        View foreGroundView = ((NoteAdapter.NoteViewHodel) viewHolder).mLayoutForceGround;
        getDefaultUIUtil().clearView(foreGroundView);
    }
}

File: app/src/main/java/com/example/project_note/RecyclerViewItemTouchHelper.kt
package com.example.project_note

import android.graphics.Canvas
import android.view.View
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.RecyclerView
import com.example.project_note.NoteAdapter.NoteViewHodel

/*
* gọi ItemTouchHelper.SimpleCallback  lắng nghe sự kiện "di chuyển" và "vuốt"
* */
class RecyclerViewItemTouchHelper    //dragDirs Hướng kéo
// swipe Dirs Hướng vuốt
    (dragDirs: Int, swipeDirs: Int, private val mlistener: ItemTouchHelperListener?) :
    ItemTouchHelper.SimpleCallback(dragDirs, swipeDirs) {
    override fun onMove(
        recyclerView: RecyclerView,
        viewHolder: RecyclerView.ViewHolder,
        target: RecyclerView.ViewHolder
    ): Boolean {
        return false
    }

    // Thông báo mListener khi swiped để thực hiện khi swip
    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
        mlistener?.onSwiped(viewHolder)
    }

    //Được gọi khi một item được chọn.
    override fun onSelectedChanged(viewHolder: RecyclerView.ViewHolder?, actionState: Int) {
        if (viewHolder != null) {
            val foreGroundView: View = (viewHolder as NoteViewHodel).mLayoutForceGround
            getDefaultUIUtil().onSelected(foreGroundView)
        }
    }

    //Được gọi khi một item được vẽ trên cùng của các item khác.
    override fun onChildDrawOver(
        c: Canvas,
        recyclerView: RecyclerView,
        viewHolder: RecyclerView.ViewHolder,
        dX: Float,
        dY: Float,
        actionState: Int,
        isCurrentlyActive: Boolean
    ) {
        val foreGroundView: View = (viewHolder as NoteViewHodel).mLayoutForceGround
        getDefaultUIUtil().onDrawOver(
            c,
            recyclerView,
            foreGroundView,
            dX,
            dY,
            actionState,
            isCurrentlyActive
        )
    }

    //Được gọi khi một item được vẽ.
    override fun onChildDraw(
        c: Canvas,
        recyclerView: RecyclerView,
        viewHolder: RecyclerView.ViewHolder,
        dX: Float,
        dY: Float,
        actionState: Int,
        isCurrentlyActive: Boolean
    ) {
        val foreGroundView: View = (viewHolder as NoteViewHodel).mLayoutForceGround
        getDefaultUIUtil().onDraw(
            c,
            recyclerView,
            foreGroundView,
            dX,
            dY,
            actionState,
            isCurrentlyActive
        )
    }

    //Được gọi khi việc vẽ item kết thúc
    override fun clearView(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder) {
        val foreGroundView: View = (viewHolder as NoteViewHodel).mLayoutForceGround
        getDefaultUIUtil().clearView(foreGroundView)
    }
}